ruby/private/bundle/create_bundle_build_file.rb

```diff
@@ -68,9 +68,10 @@
 #
 # Library path differs across implementations as `lib/ruby` on MRI and `lib/jruby` on JRuby.
 GEM_PATH = ->(ruby_version, gem_name, gem_version) do
-  glob = Dir.glob("lib/#{RbConfig::CONFIG['RUBY_INSTALL_NAME']}/#{ruby_version}/gems/#{gem_name}-#{gem_version}*").first
```

Comment:
The upstream repo that this repo forked stull uses [`RUBY_INSTALL_NAME`](https://github.com/bazelruby/rules_ruby/blob/master/ruby/private/bundle/create_bundle_build_file.rb#L71), which suggests that for most users, `RUBY_INSTALL_NAME` works as expected. Do you know why this doesn't work in your environment, e.g. how do these two values differ on your system?

Reply:
The relevant parts of my dev environment:
![image](https://github.com/user-attachments/assets/b022dbf2-3da5-49cf-8a82-8c1b1cd037a8)

As you can see, it's debian `trixie` running the default `ruby` package.

If you look at the output of running `RbConfig::MAKEFILE_CONFIG` in irb on my machine [here](https://gist.github.com/tempoz/988131e8eede397ec4a304416edf71a2), and the output for `2.3` that's posted on the [idiosyncratic ruby site I posted above](https://idiosyncratic-ruby.com/42-ruby-config.html#rbconfigconfig), you can see that the difference is that in `2.3` `RUBY_INSTALL_NAME` is equivalent to `RUBY_BASE_NAME`, whereas in `3.1` `RUBY_INSTALL_NAME` is `RUBY_BASE_NAME` with the string `3.1` appended. Since `rubylibprefix` is defined as `$(libdir)/$(RUBY_BASE_NAME)` (in both versions), it was fine to just substitute `RUBY_INSTALL_NAME` in `2.3` when the strings were equivalent, but now that they aren't, we have to use `RUBY_BASE_NAME`.

As an aside, I think it would be equally correct to use `rubylibprefix` and trim `libdir` from the beginning, but this is simpler.